### ISSUE TITLE
Change detection stops working when using timers in web workers

### ISSUE THREAD

**Issue by thelgevold** [angular/angular#9201]
* **I'm submitting a ...**
- [x] bug report
- [ ] feature request
- [ ] support request => Please do not submit support request here, see note at the top of this template.

**Current behavior**
Change Detection seems to stop working when updates are made in a ```setTimeout``` running inside a web worker. It works initially, but as the timer ticks away, at some point it stops updating the UI. The setTimeout itself does however not stop - just the UI updates
**Expected/desired behavior**
It should keep updating the UI for as long as the timer is active.

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem** via
Here is a plunker with a repro of the issue: http://plnkr.co/edit/SPwWzK8f9ifUnwiUGveS?p=preview

You may have to keep it running for a little while, but for me it happens relatively quickly.

* **Please tell us about your environment:**

* **Angular version:** 2.0.0-master 

Browser: Chrome 52

**Comment by vicb** [comment]
you're missing zone.js worker side. re-open if it does not fix your issue.

**Comment by thelgevold** [comment]
Ok. I will try that. Curious why it works initially though and stops working after a while...

**Comment by thelgevold** [comment]
@vicb I updated the Plunker, but the issue is still there. Can you confirm that the setup is correct in the loader?

**Comment by vicb** [comment]
http://plnkr.co/edit/PdYDuxy5sUjn7Yt5F7XL?p=preview

worker related stuff are back in browser-dynamic... I can agree it's hard to follow :)

**Comment by thelgevold** [comment]
Hm.. It looks like the updated plunker has the same problem. The time it takes before it stops updating might vary based on hardware etc, but in my case it stops updating the "clock" if I leave it running for 30 sec...  

For some reason it fails silently though...

Tried updating my actual project as well, but with no luck there either...

Does it not break for you?

**Comment by thelgevold** [comment]
@vicb I am not convinced this issue should be closed..

I updated the plunker to rc 2 to include the last minute api changes here http://plnkr.co/edit/7ycV3GzAqXmvpyOMTKIl?p=preview

Still reproducible. The timer stops working after a while....

**Comment by vicb** [comment]
What is a while ? I can't reproduce

**Comment by thelgevold** [comment]
It varies from time to time, but for me it happens within a minute for sure.

Using Chrome 52 on OSX

**Comment by vicb** [comment]
Any console logs ?

**Comment by thelgevold** [comment]
That's the interesting part. It fails silently, but the console.log still keeps on logging. Even though the UI stops updating. 
I believe @StephenFluin was able to repro it as well yesterday. 

A bit weird :-)

**Comment by cyrfer** [comment]
Text stopped updating after ~20 seconds for me. Windows 10 x64, Chrome Version 51.0.2704.84 m.

**Comment by thelgevold** [comment]
Updated the plunker to rc4. Currently it seems like the issue is only reproducible in Chrome

Firefox and IE edge seem fine. However, Chrome 52.0.2743.49 beta (64-bit) still displays the same issue where the timer stops working after 20-30 seconds.

**Comment by JurajMlich** [comment]
Mac Chrome 51.0.2704.103 (64-bit) reproducible
